src/backend/packages/inensus/spark-meter/src/Observers/GeographicalInformationObserver.php

```diff
@@ -20,36 +21,72 @@ public function __construct(
     ) {}
 
     public function updated(GeographicalInformation $geographicalInformation) {
-        if ($this->pluginsService->isPluginActive(MpmPlugin::SPARK_METER) && $geographicalInformation->owner_type === 'meter_parameter') {
-            $meterId = $geographicalInformation->owner_id;
-            $customer = $this->person->newQuery()->with(['meters.tariff', 'meters.geo', 'meters.meter'])
-                ->whereHas('meters', function ($q) use ($meterId) {
-                    return $q->where('id', $meterId);
-                })->first();
-            $smCustomer = $this->smCustomer->newQuery()
-                ->where('mpm_customer_id', $customer->id)->first();
-            if ($smCustomer) {
-                $address = $customer->addresses()->where('is_primary', 1)->first();
-                $customerData = [
-                    'id' => $smCustomer->customer_id,
-                    'active' => true,
-                    'meter_tariff_name' => $customer->meters[0]->tariff->name,
-                    'name' => $customer->name.' '.$customer->surname,
-                    'code' => strval($customer->id),
-                    'phone_number' => $address->phone,
-                    'coords' => $customer->meters[0]->geo->points,
-                    'address' => $address->street,
-                ];
-
-                $this->customerService->updateSparkCustomerInfo($customerData, $smCustomer->site_id);
-                $smModelHash = $this->customerService->hashCustomerWithMeterSerial(
-                    $customer->meters[0]->meter->serial_number,
-                    $smCustomer->site_id
-                );
-                $smCustomer->update([
-                    'hash' => $smModelHash,
-                ]);
+        if (!$this->pluginsService->isPluginActive(MpmPlugin::SPARK_METER)) {
+            return;
+        }
+
+        if ($geographicalInformation->owner_type === 'address') {
+            $address = $geographicalInformation->owner;
+
+            if ($address && $address->owner_type === 'device') {
+                $device = $address->owner;
+
+                if ($device && $device->device_type === 'meter') {
+                    $this->updateSparkMetaCustomerInformation($device, $geographicalInformation);
+                }
             }
         }
     }
+
+    /**
+     * Update Spark meter customer information.
+     *
+     * @param Device                  $device
+     * @param GeographicalInformation $geographicalInformation
+     *
+     * @return void
+     */
+    private function updateSparkMetaCustomerInformation(Device $device, GeographicalInformation $geographicalInformation) {
+        $meter = $device->device;
+
+        $customer = $this->person->newQuery()
+            ->with(['meters.tariff', 'meters.geo', 'meters.meter'])
```

Comment:
I don't think this works anymore. Here we have to replace the `meters` relation with the `devices.device` relation.